docs/source/jit_language_reference_statements.rst

```diff
@@ -0,0 +1,389 @@
+    for x in tup:
+        print(x)
+
+*  for loops on lists: For loops over a ``nn.ModuleList`` will unroll the body of the loop at compile time,
+    * with each member of the module list.
+
+    For eg.,
+
+    class SubModule(torch.nn.Module):
+        def __init__(self):
+            super(SubModule, self).__init__()
+            self.weight = nn.Parameter(torch.randn(2))
+
+        def forward(self, input):
+            return self.weight + input
+
+    class MyModule(torch.nn.Module):
+
+        def __init__(self):
+            super(MyModule, self).init()
+            self.mods = torch.nn.ModuleList([SubModule() for i in range(10)])
+
+        def forward(self, v):
+            for module in self.mods:
+                v = module(v)
+            return v
+
+    model = torch.jit.script(MyModule())
+
+The ``with`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^
+* The `with <https://docs.python.org/3/reference/compound_stmts.html#with>` statement is used to wrap the
+* execution of a block with methods defined by a context manager
+
+::
+    with_stmt ::=  "with" with_item ("," with_item) ":" suite
+    with_item ::=  expression ["as" target]
+    `with_item <https://docs.python.org/3/reference/compound_stmts.html#grammar-token-with-item>`
+    `suite <https://docs.python.org/3/reference/compound_stmts.html#grammar-token-suite>`
+
+* If a target was included in the `with <https://docs.python.org/3/reference/compound_stmts.html#with>` statement,
+* the return value from context manager’s `__enter__() <https://docs.python.org/3/reference/datamodel.html#object.__enter__>`
+* is assigned to it. If an exception caused the suite to be exited, its type, value, and traceback are passed as
+* arguments to `__exit__() <https://docs.python.org/3/reference/datamodel.html#object.__exit__>. Otherwise, three
+* `None <https://docs.python.org/3/library/constants.html#None>` arguments are supplied.
+
+* `try/except/finally` statements are not supported inside `with` blocks.
+*  Exceptions raised within `with` block cannot be suppressed.
+
+The ``tuple`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^
+
+::
+    tuple_stmt ::= tuple([iterables])
+
+* Iterable types in TorchScript include `Tensors`, `lists`, `tuples`, `dictionaries`, `strings`,
+    * `torch.nn.ModuleList` and `torch.nn.ModuleDict`.
+* Cannot convert a List to Tuple by using this built-in function.
+* Unpacking all outputs into a tuple is covered by:
+    abc = func() # Function that returns a tuple
+    a,b = func()
+
+The ``getattr`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    getattr_stmt ::= getattr(object, name[, default])
+
+* Attribute name must be a literal string.
+* Module type object is not supported for eg., torch._C
+* Custom Class object is not supported for eg., torch.classes.*
+
+The ``getattr`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+::
+    hasattr_stmt ::= hasattr(object, name)
+
+* Attribute name must be a literal string.
+* Module type object is not supported for eg., torch._C
```

Comment:
I assume `Module` here refers to Python modules? If so, could we make it clearer? Also, it might be better to use an example other than `torch._C` which is not exposed to users and they may not know what it is at all.

Reply:
We only support on `self` objects and not on any other module type objects. May be I can add this clarification here.